SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -702,47 +704,56 @@ void USpatialActorChannel::UpdateSpatialPosition()
 {
 	SCOPE_CYCLE_COUNTER(STAT_SpatialActorChannelUpdateSpatialPosition);
 
-	// PlayerController's and PlayerState's are a special case here. To ensure that they and their associated pawn are 
-	// handed between workers at the same time (which is not guaranteed), we ensure that we update the position component 
-	// of the PlayerController and PlayerState at the same time as the pawn.
+	// When we update an Actor's position, we want to update the position of all the children of this Actor.
+	// If this Actor is a PlayerController, we want to update all of its children and its possessed Pawn.
+	// That means if this Actor has an Owner or has a NetConnection and is NOT a PlayerController
+	// we want to defer updating position until we reach the highest parent.
+	if ((Actor->GetOwner() != nullptr || Actor->GetNetConnection() != nullptr) && !Actor->IsA<APlayerController>())
+	{
+		return;
+	}
+
+	// Check that there has been a sufficient amount of time since the last update.
+	if ((NetDriver->Time - TimeSinceLastPositionUpdate) < (1.0 / GetDefault<USpatialGDKSettings>()->PositionUpdateFrequency))
+	{
+		return;
+	}
 
-	// Check that it has moved sufficiently far to be updated
-	const float SpatialPositionThreshold = 100.0f * 100.0f; // 1m (100cm)
+	// Check that the Actor has moved sufficiently far to be updated
+	const float SpatialPositionThresholdSquared = FMath::Square(GetDefault<USpatialGDKSettings>()->PositionDistanceThreshold);
 	FVector ActorSpatialPosition = GetActorSpatialPosition(Actor);
-	if (FVector::DistSquared(ActorSpatialPosition, LastSpatialPosition) < SpatialPositionThreshold)
+	if (FVector::DistSquared(ActorSpatialPosition, LastPositionSinceUpdate) < SpatialPositionThresholdSquared)
 	{
 		return;
 	}
 
-	LastSpatialPosition = ActorSpatialPosition;
-	Sender->SendPositionUpdate(EntityId, LastSpatialPosition);
+	LastPositionSinceUpdate = ActorSpatialPosition;
+	TimeSinceLastPositionUpdate = NetDriver->Time;
```

Comment:
The name `TimeSinceLastPositionUpdate` implies this is a time interval but it's actually a time point.

Reply:
Can change.